src/features/Lock/lock.js

```diff
@@ -58,6 +59,8 @@ const Lock = () => {
   const [currentHiIQ, setCurrentHiIQ] = useState(undefined);
   const [filledAmount, setFilledAmount] = useState();
   const [openWrongChainModal, setOpenWrongChainModal] = useState(false);
+  const [openTxDetailsDialog, setOpenTxDetailsDialog] = useState(true);
+  const [hashes, setHashes] = useState([]);
```

Comment:
this is going to be repeated more often, I think it could be a good moment to create a custom hook for it :)

Reply:
@kesar  you mean   `const [openTxDetailsDialog, setOpenTxDetailsDialog] = useState(true);` statement?

Reply:
no I meant to create a useTransaction() that might encapsulate all this logic to avoid duplicity. 

We could also probably move the dialog to the layout since its gonna be reused across most likely all the views 👍🏻 

Feel free to create issues for refactoring so we can merge this one and improve it future iterations